iphone/Maps/Classes/MapsAppDelegate.mm

```diff
@@ -222,12 +226,19 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
     f.OnRecoverSurface(static_cast<int>(objcSize.width), static_cast<int>(objcSize.height),
                        true /* recreateContextDependentResources */);
   }
+  [self startWatchingClipboard];
   [MWMLocationManager applicationDidBecomeActive];
   [MWMSearch addCategoriesToSpotlight];
   [MWMKeyboard applicationDidBecomeActive];
   [MWMTextToSpeech applicationDidBecomeActive];
   LOG(LINFO, ("applicationDidBecomeActive - end"));
 }
+- (void)startWatchingClipboard {
+    NSLog(@"Watching started");
+  ClipboardWatcher *clipboardWatcher = [[ClipboardWatcher alloc] init];
+  UIWindow *targetWindow = [UIApplication sharedApplication].keyWindow;
+  [clipboardWatcher startWatching:targetWindow];
```

Comment:
Is watching really needed? Can pasteboard checking be implemented only on the app activation?

Reply:
It does only check the pasteboard once when the app becomes active, I have changed the name to clipboardChecker to avoid any misunderstanding. Also it is called under applicationDidBecomeActive() of MapsAppDelegate.mm